debezium-core/src/main/java/io/debezium/relational/mapping/MaskStrings.java

```diff
@@ -128,9 +130,8 @@ public HashValueConverter(byte[] salt, String hashAlgorithm) {
         public Object convert(Object value) {
             if (value instanceof Serializable) {
                 try {
-                    return toHash((Serializable) value);
-                }
-                catch (IOException e) {
+                    return Version.V1.equals(version) ? toHash((Serializable) value) : toHashV2((Serializable) value);
```

Comment:
Let's push the actual hashing implementation into a method on enum members themselves and here simply call:

    hashingStrategy.toHash(value);